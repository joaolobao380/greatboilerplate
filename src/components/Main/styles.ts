import styled from 'styled-components';

export const Wrapper = styled.main`
  background-color: #19191c;
  color: #ffffff;
  width: 100%;
  height: 100%;
  justify-content: center;
  align-items: center;
  display: flex;
  flex-direction: column;
`;

export const Logo = styled.img`
  margin-bottom: 2rem;
`;

export const Title = styled.h1`
  font-family: Rowdies;
  text-align: center;
  font-size: 4rem;
`;

export const Description = styled.h3`
  text-align: center;
  margin-left: 1rem;
  margin-right: 1rem;
  margin-top: 1.5rem;
  font-size: 2rem;
`;

export const GitHub = styled.div`
  margin-top: 1rem;
  cursor: pointer;
`;

export const GitHubImg = styled.img`
  margin-right: 0.5rem;
`;

export const GitHubText = styled.span`
  font-size: 1.5rem;
  font-family: Rowdies;
`;

export const Linkedin = styled.div`
  margin-top: 0.5rem;
  cursor: pointer;
`;

export const LinkedinImg = styled.img`
  margin-right: 0.5rem;
`;

export const LinkedinText = styled.span`
  font-size: 1.5rem;
  font-family: Rowdies;
`;

export const Author = styled.h4`
  margin-top: 1.5rem;
  font-size: 1.5rem;
  font-family: Rowdies;
`;
