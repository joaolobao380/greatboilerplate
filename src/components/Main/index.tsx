import Link from 'next/link';
import * as S from './styles';

const Main = ({
  title = 'Great Boilerplate!',
  description = 'NextJs, ReactJs, TypeScript, Eslint, Prettier, Styled components,Test library, Husky, Lint staged, Storybook',
  author = 'João Victor Lobão',
}) => (
  <S.Wrapper>
    <S.Logo src="/img/Logo.svg" alt="Bo, boilerplate graphic art" />
    <S.Title>{title}</S.Title>
    <S.Description>{description}</S.Description>
    <Link href="https://github.com/joaolobao380" replace>
      <S.GitHub>
        <S.GitHubImg src="/img/GithubImg.svg" alt="GitHub image" />
        <S.GitHubText>GitHub</S.GitHubText>
      </S.GitHub>
    </Link>
    <Link href="https://www.linkedin.com/in/jo%C3%A3o-victor-lob%C3%A3o-1b6b5012a/">
      <S.Linkedin>
        <S.LinkedinImg src="/img/LinkedinImg.svg" alt="Linkedin image" />
        <S.LinkedinText>Linkedin</S.LinkedinText>
      </S.Linkedin>
    </Link>
    <S.Author>Created by {author}</S.Author>
  </S.Wrapper>
);

export default Main;
