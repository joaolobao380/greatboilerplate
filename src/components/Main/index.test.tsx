import { render, screen } from '@testing-library/react';
import Main from '.';

describe('<Main />', () => {
  it('should render the heading', () => {
    const { container } = render(<Main />);

    expect(
      screen.getByRole('heading', { name: /Great Boilerplate!/i })
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  //Testing styles
  it('should render the background color correctly', () => {
    const { container } = render(<Main />);

    expect(container.firstChild).toHaveStyle({ 'background-color': '#19191c' });
  });
});
