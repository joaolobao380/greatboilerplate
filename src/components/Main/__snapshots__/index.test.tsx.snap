// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
<main
  class="sc-bdnxRM hPiJmr"
>
  <img
    alt="Bo, boilerplate graphic art"
    class="sc-gtsrHT kfulkT"
    src="/img/Logo.svg"
  />
  <h1
    class="sc-dlnjwi iAWOCh"
  >
    Great Boilerplate!
  </h1>
  <h3
    class="sc-hKFxyN lgZNsR"
  >
    NextJs, ReactJs, TypeScript, Eslint, Prettier, Styled components,Test library, Husky, Lint staged, Storybook
  </h3>
  <div
    class="sc-eCApnc cgWuHY"
  >
    <img
      alt="GitHub image"
      class="sc-jSFjdj kgyLI"
      src="/img/GithubImg.svg"
    />
    <span
      class="sc-gKAaRy kLGZYt"
    >
      GitHub
    </span>
  </div>
  <div
    class="sc-iCoGMd dqloiT"
  >
    <img
      alt="Linkedin image"
      class="sc-fujyAs eqCsdt"
      src="/img/LinkedinImg.svg"
    />
    <span
      class="sc-pNWdM fJoMZW"
    >
      Linkedin
    </span>
  </div>
  <h4
    class="sc-jrsJWt gFbCNB"
  >
    Created by 
    João Victor Lobão
  </h4>
</main>
`;
